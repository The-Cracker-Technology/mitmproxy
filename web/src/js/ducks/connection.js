export const ConnectionState = {
    INIT: Symbol("init"),
    FETCHING: Symbol("fetching"), // WebSocket is established, but still fetching resources.
    ESTABLISHED: Symbol("established"),
    ERROR: Symbol("error"),
    OFFLINE: Symbol("offline"), // indicates that there is no live (websocket) backend.
}

const defaultState = {
    state: ConnectionState.INIT,
    message: null,
}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case ConnectionState.ESTABLISHED:
        case ConnectionState.FETCHING:
        case ConnectionState.ERROR:
        case ConnectionState.OFFLINE:
            return {
                state: action.type,
                message: action.message
            }

        default:
            return state
    }
}

export function startFetching() {
    return { type: ConnectionState.FETCHING }
}

export function connectionEstablished() {
    return { type: ConnectionState.ESTABLISHED }
}

export function connectionError(message) {
    return { type: ConnectionState.ERROR, message }
}
export function setOffline() {
    return { type: ConnectionState.OFFLINE }
}
