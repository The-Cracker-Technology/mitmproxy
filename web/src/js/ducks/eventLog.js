import reduceStore from "./utils/store"
import * as storeActions from "./utils/store"

export const ADD               = 'EVENTS_ADD'
export const RECEIVE           = 'EVENTS_RECEIVE'
export const TOGGLE_VISIBILITY = 'EVENTS_TOGGLE_VISIBILITY'
export const TOGGLE_FILTER     = 'EVENTS_TOGGLE_FILTER'

const defaultState = {
    visible: false,
    filters: { debug: false, info: true, web: true, warn: true, error: true },
    ...reduceStore(undefined, {}),
}

export default function reduce(state = defaultState, action) {
    switch (action.type) {

        case TOGGLE_VISIBILITY:
            return {
                ...state,
                visible: !state.visible
            }

        case TOGGLE_FILTER:
            const filters = { ...state.filters, [action.filter]: !state.filters[action.filter] }
            return {
                ...state,
                filters,
                ...reduceStore(state, storeActions.setFilter(log => filters[log.level]))
            }

        case ADD:
        case RECEIVE:
            return {
                ...state,
                ...reduceStore(state, storeActions[action.cmd](action.data, log => state.filters[log.level]))
            }

        default:
            return state
    }
}

export function toggleFilter(filter) {
    return { type: TOGGLE_FILTER, filter }
}

export function toggleVisibility() {
    return { type: TOGGLE_VISIBILITY }
}

export function add(message, level = 'web') {
    let data = {
        id: Math.random().toString(),
        message,
        level,
    }
    return {
        type: ADD,
        cmd: "add",
        data
    }
}
