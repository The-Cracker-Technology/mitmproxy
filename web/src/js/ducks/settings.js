import { fetchApi } from '../utils'

export const RECEIVE        = 'SETTINGS_RECEIVE'
export const UPDATE         = 'SETTINGS_UPDATE'
export const REQUEST_UPDATE = 'REQUEST_UPDATE'

const defaultState = {

}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case RECEIVE:
            return action.data

        case UPDATE:
            return {
                ...state,
                ...action.data,
            }

        default:
            return state
    }
}

export function update(settings) {
    fetchApi.put('/settings', settings)
    return { type: REQUEST_UPDATE }
}
