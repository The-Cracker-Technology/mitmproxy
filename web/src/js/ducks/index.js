import { combineReducers } from "redux"
import eventLog from "./eventLog"
import flows from "./flows"
import settings from "./settings"
import ui from "./ui/index"
import connection from "./connection"
import options from './options'

export default combineReducers({
    eventLog,
    flows,
    settings,
    connection,
    ui,
    options,
})
