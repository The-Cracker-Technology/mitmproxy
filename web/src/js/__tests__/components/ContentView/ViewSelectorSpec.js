import React from 'react'
import renderer from 'react-test-renderer'
import ConnectedComponent, { ViewSelector } from '../../../components/ContentView/ViewSelector'
import { Provider } from 'react-redux'
import { TStore } from '../../ducks/tutils'


describe('ViewSelector Component', () => {
    let contentViews = ['Auto', 'Raw', 'Text'],
        activeView = 'Auto',
        setContentViewFn = jest.fn(),
        viewSelector = renderer.create(
            <ViewSelector contentViews={contentViews} activeView={activeView} setContentView={setContentViewFn}/>
        ),
        tree = viewSelector.toJSON()

    it('should render correctly', () => {
        expect(tree).toMatchSnapshot()
    })

    it('should handle click', () => {
        let mockEvent = { preventDefault: jest.fn() },
            tab = tree.children[1].children[0].children[1]
        tab.props.onClick(mockEvent)
        expect(mockEvent.preventDefault).toBeCalled()
    })

    it('should connect to state', () => {
        let store = TStore(),
            provider = renderer.create(
                <Provider store={store}>
                    <ConnectedComponent/>
                </Provider>
            ),
            tree = provider.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
