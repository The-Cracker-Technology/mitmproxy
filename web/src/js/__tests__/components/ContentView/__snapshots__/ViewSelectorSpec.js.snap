// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSelector Component should connect to state 1`] = `
<div
  className="dropup pull-left"
>
  <a
    className="btn btn-default btn-xs"
    href="#"
    onClick={[Function]}
  >
    <span>
       
      <b>
        View:
      </b>
       
      auto
       
      <span
        className="caret"
      />
       
    </span>
  </a>
  <ul
    className="dropdown-menu"
    role="menu"
  >
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        auto
      </a>
       
    </li>
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        raw
      </a>
       
    </li>
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        text
      </a>
       
    </li>
  </ul>
</div>
`;

exports[`ViewSelector Component should render correctly 1`] = `
<div
  className="dropup pull-left"
>
  <a
    className="btn btn-default btn-xs"
    href="#"
    onClick={[Function]}
  >
    <span>
       
      <b>
        View:
      </b>
       
      auto
       
      <span
        className="caret"
      />
       
    </span>
  </a>
  <ul
    className="dropdown-menu"
    role="menu"
  >
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        auto
      </a>
       
    </li>
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        raw
      </a>
       
    </li>
    <li>
       
      <a
        href="#"
        onClick={[Function]}
      >
        text
      </a>
       
    </li>
  </ul>
</div>
`;
